app/integration/component-proxy/src/main/java/io/syndesis/integration/component/proxy/ComponentCustomizer.java

```diff
@@ -44,23 +44,37 @@ default void consumeOption(Map<String, Object> options, String name, Consumer<Ob
         }
     }
 
+    default void consumeStringOption(Map<String, Object> options, String name, Consumer<String> consumer) {
+        Object val = options.remove(name);
+        if (val != null) {
+            consumer.accept(String.valueOf(val));
+        }
+    }
+
     /**
      * Removes the option with the given name and invokes the consumer performing
      * property placeholders resolution and type conversion.
      */
-    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
-    default <T> void consumeOption(CamelContext camelContext, Map<String, Object> options, String name, Class<T> type, Consumer<T> consumer) throws Exception {
-        TypeConverter converter = camelContext.getTypeConverter();
+    default <T> void consumeOption(CamelContext camelContext, Map<String, Object> options, String name, Class<T> type, Consumer<T> consumer) {
         Object val = options.remove(name);
 
         if (val != null) {
-            if (val instanceof String) {
-                val = camelContext.resolvePropertyPlaceholders((String)val);
-            }
+            try {
+                if (val instanceof String) {
+                    val = camelContext.resolvePropertyPlaceholders((String)val);
+                }
 
-            consumer.accept(
-                converter.mandatoryConvertTo(type, val)
-            );
+                if (type.isInstance(val)) {
```

Comment:
this does not hurt but I think the type converter does that by default

Reply:
No need to invoke one if not needed, that was the idea...

Reply:
yes but it is the same check the type converter does, anyway not an issue